src/common/personinfo.js

```diff
@@ -167,8 +167,34 @@ let personinfo = function (tour, aConfig) {
   });
 };
 
+
+const PERSISTENT_LOCALSTORAGE_KEY="localStorageUsed";
+let tryLocalStorage = function (store=localStorage) {
+  try {
+    var haveChecked = JSON.parse(store.getItem(PERSISTENT_LOCALSTORAGE_KEY) || 0);
+    if (!haveChecked || haveChecked < 2) {
+      store.setItem(PERSISTENT_LOCALSTORAGE_KEY, ++haveChecked);
+      return;
+    }
+  } catch (e) {
+    store.setItem(PERSISTENT_LOCALSTORAGE_KEY, 1);
+  }
+};
+
+let isLocalStoragePersistent = function (store=localStorage) {
+  var haveChecked = store.getItem(PERSISTENT_LOCALSTORAGE_KEY) || "0";
+  try {
+    return JSON.parse(haveChecked) >= 2;
```

Comment:
Made consistent at least.  

I am open to suggestions about what is an error here.  It's definitely hacky.